test/Microsoft.ML.Tests/Scenarios/IrisPlantClassificationWithStringLabelTests.cs

```diff
@@ -3,126 +3,100 @@
 
-            Assert.Equal(1, prediction.PredictedLabels[0], 2);
-            Assert.Equal(0, prediction.PredictedLabels[1], 2);
-            Assert.Equal(0, prediction.PredictedLabels[2], 2);
+            Assert.Equal(1, prediction.PredictedScores[0], 2);
+            Assert.Equal(0, prediction.PredictedScores[1], 2);
+            Assert.Equal(0, prediction.PredictedScores[2], 2);
+            Assert.True(prediction.PredictedPlant == "Iris-setosa");
 
-            prediction = model.Predict(new IrisDataWithStringLabel()
+            prediction = predictFunction.Predict(new IrisDataWithStringLabel()
             {
                 SepalLength = 6.4f,
                 SepalWidth = 3.1f,
                 PetalLength = 5.5f,
                 PetalWidth = 2.2f,
             });
 
-            Assert.Equal(0, prediction.PredictedLabels[0], 2);
-            Assert.Equal(0, prediction.PredictedLabels[1], 2);
-            Assert.Equal(1, prediction.PredictedLabels[2], 2);
+            Assert.Equal(0, prediction.PredictedScores[0], 2);
+            Assert.Equal(0, prediction.PredictedScores[1], 2);
+            Assert.Equal(1, prediction.PredictedScores[2], 2);
+            Assert.True(prediction.PredictedPlant == "Iris-virginica");
 
-            prediction = model.Predict(new IrisDataWithStringLabel()
+            prediction = predictFunction.Predict(new IrisDataWithStringLabel()
             {
                 SepalLength = 4.4f,
                 SepalWidth = 3.1f,
                 PetalLength = 2.5f,
                 PetalWidth = 1.2f,
             });
 
-            Assert.Equal(.2, prediction.PredictedLabels[0], 1);
-            Assert.Equal(.8, prediction.PredictedLabels[1], 1);
-            Assert.Equal(0, prediction.PredictedLabels[2], 2);
+            Assert.Equal(.2, prediction.PredictedScores[0], 1);
+            Assert.Equal(.8, prediction.PredictedScores[1], 1);
+            Assert.Equal(0, prediction.PredictedScores[2], 2);
+            Assert.True(prediction.PredictedPlant == "Iris-versicolor");
 
-            // Note: Testing against the same data set as a simple way to test evaluation.
-            // This isn't appropriate in real-world scenarios.
-            string testDataPath = GetDataPath("iris.data");
-            var testData = new TextLoader(testDataPath).CreateFrom<IrisDataWithStringLabel>(useHeader: false, separator: ',');
-
-            var evaluator = new ClassificationEvaluator();
-            evaluator.OutputTopKAcc = 3;
-            ClassificationMetrics metrics = evaluator.Evaluate(model, testData);
+            // Evaluate the trained pipeline
+            var predicted = trainedModel.Transform(testData);
+            var metrics = mlContext.MulticlassClassification.Evaluate(predicted, topK: 3);
 
             Assert.Equal(.98, metrics.AccuracyMacro);
             Assert.Equal(.98, metrics.AccuracyMicro, 2);
-            Assert.Equal(.06, metrics.LogLoss, 2);
+            Assert.InRange(metrics.LogLoss, .05, .06);
```

Comment:
Assert.Equal(metrics.Logloss, 0.055, 3) maybe?

Reply:
fixed

---
In reply to: [245133430](https://github.com/dotnet/machinelearning/pull/2008#discussion_r245133430) [](ancestors = 245133430)